source/core/parameters.cc

```diff
@@ -1548,6 +1558,39 @@ namespace Parameters
                           "velocity|pressure|phase|temperature"),
                         "Variable for kelly estimation"
                         "Choices are <velocity|pressure|phase|temperature>.");
+      // end WIP
+      // WIP declare multivariable parameters
+      prm.declare_entry(
+        "types",
```

Comment:
There will be only one type of refinement and not multiple. For a given variable you have kelly, none, uniform. If you have a uniform for one variable you will have the equivalent of a uniform for all variables. If you have none for a given variable you will just not have this variable in the list of variables. I would keep the original type of refinement.

Reply:
and well, I still need to implement the "uniform" type, for it to be usable also :)

Reply:
I would rather keep one type (kelly, uniform, none) for all variables. Otherwise it will get confusing. I understand what you mean about turning on or off, but this will be better accomplished by commenting / uncommenting the 3-4 lines that defines the variables and their parameter instead.